packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -116,6 +142,69 @@ export const AlephiumConnectProvider: React.FC<{ children?: React.ReactNode }> =
   return <AlephiumConnectContext.Provider value={value}>{children}</AlephiumConnectContext.Provider>
 }
 
+export const AlephiumBalanceProvider: React.FC<{ children?: React.ReactNode }> = ({ children }) => {
+  const context = useContext(AlephiumBalanceContext)
+  if (context) {
+    throw new Error('Multiple, nested usages of AlephiumBalanceProvider detected. Please use only one.')
+  }
+
+  const { account, signerProvider } = useAlephiumConnectContext()
+  const [balance, setBalance] = useState<node.Balance | undefined>()
+
+  const updateBalance = useCallback(async () => {
+    const nodeProvider = tryGetNodeProvider() ?? signerProvider?.nodeProvider
+    if (nodeProvider && account) {
+      const newBalance = await nodeProvider.addresses.getAddressesAddressBalance(account.address)
+      setBalance((prevBalance) => {
+        if (prevBalance !== undefined && isBalanceEqual(prevBalance, newBalance)) {
+          return prevBalance
+        }
+        return newBalance
+      })
+    } else if (account === undefined) {
+      setBalance(undefined)
+    }
+  }, [account, signerProvider])
+
+  const updateBalanceForTx = useCallback(
+    (txId: string, confirmations?: number) => {
```

Comment:
Let's throw an error when `account === undefined`. One should not call `updateBalanceForTx` when account is not available